### DISCUSSION TITLE
Pagination Width Limit

### DISCUSSION THREAD

**Discussion by StormNye** [Legal-and-General/canopy#1431] [Canopy requests]
I have a potential improvement for the pagination component.

The pagination component has an issue where it will exceed the edges of the page if there are too many items:
<img width="1251" alt="Screenshot 2024-07-16 at 16 36 22" src="https://github.com/user-attachments/assets/70e5f460-7d0a-4cad-9941-a2180b9ce98e">
Note that you can scroll right, but you can't scroll left to the earlier pages, making it unusable.

This depends on screen size, number of items and items per page. I've written something that will limit it to the available screen width:
<img width="1309" alt="Screenshot 2024-07-16 at 16 35 50" src="https://github.com/user-attachments/assets/c00baaff-6d36-4667-8083-10f7a91d5536">

It scales down to mobile sizes well:
<img width="408" alt="Screenshot 2024-07-16 at 16 40 10" src="https://github.com/user-attachments/assets/3f866438-2200-4294-9996-49e6956b8774">

- It only comes into effect when the number of pages exceeds the screen width.
- It deals with the current issue with mobile sizes listed under additional notes in the documentation.
- It uses the goto start/end icons shown in the previous pagination design discussion.
- It should deal with changes to the base font size that would change the size of the icons and page numbers within.

Here is the branch:
https://github.com/StormNye/StormNye-canopy/tree/pagination-width-limit

Could I get some feedback on if this would be suitable please?

Thanks

**Comment by owensgit** [comment]
Hi @StormNye, thanks for putting this together. I think it's a good idea and deals with the issue well. I'd be happy to proceed with this as a solution as long as the UX designers agree. What are your thoughts @alexcanning?

**Comment by alexcanning** [comment]
Hi @StormNye @owensgit. As I recall, the pagination component was intended to have overflow dots when the number of pages exceeded the available space, but that functionality was never added. This is how it was intended to look:

![image](https://github.com/user-attachments/assets/15c43070-a18c-4fc0-85fd-a371ca2a566b)

There was also a version of the designs at one point that had an input box so you could manually type the number you wanted to jump to.

Not saying either of these is the solution - what you've put together looks sensible. We just need to do a bit of design validation.

**Reply by alexcanning** [reply]
An example of the pagination truncation can be seen on this Atlassian component https://atlassian.design/components/pagination/examples
